### ISSUE TITLE
Reproducing chem transfer learning results and typos in code

### ISSUE THREAD

**Issue by JPaulYang** [LiuChuang0059/StructMAE#1]
Hi,

Thanks for sharing your work! I'm recently studying this field and found this work particularly interesting. So I followed the same set-up as in the code of StructMAE-L and StructMAE-P, and then trained the model for 100 epochs. Then, I saved the model weight at the 100-th epoch for transfer learning over those MoleculeNet datasets. However, I face some challenges in reproducing the results as reported in the paper, so I am wondering if there is anything missing in my reproduction. In particular, I have the following questions:

1. When selecting the best checkpoint for downstream tasks, did you apply additional criteria beyond loss convergence or validation metrics?
2. The finetuning stage seems to be unstable sometimes, do you make any additional criteria in the finetune script to help stablizing the performance (e.g. prevent overfitting?).

Also there is a minor typo in your code [here](https://github.com/LiuChuang0059/StructMAE/blob/main/StructMAE-L/chem/pretraining.py) on line 274-275. I think you should uncomment 274 and comment out 275, otherwise the node will be masked twice in the training.

Your insights would be greatly appreciated. Thanks again for your time and help!

**Comment by littlewyy** [comment]
Thank you for raising these issues. Here are our responses:

1. For selecting the best checkpoint for downstream tasks, we primarily focused on loss convergence and did not apply additional criteria.

2. The fine-tuning stage can sometimes be unstable, so we occasionally use early stopping to prevent overfitting.

3. Regarding the code issue you mentioned, the original code does not actually cause a node to be masked twice during training. This is because the relevant code for node masking in the `MaskAtom` function within the StructMAE-L project has already been commented. You can refer to [this section of code](https://github.com/LiuChuang0059/StructMAE/blob/main/StructMAE-L/chem/util.py) from lines 212-296. I realize my code may have caused some misunderstanding, so I appreciate you pointing this out!

**Comment by JPaulYang** [comment]
Thanks a lot for the reply! I realize I did not use exactly the same script for the MaskAtom part. I will try again with the original setting.
